File: ecommerce-ts-express/packages/contracts/src/orders/order.dto.ts
import { Currency } from '../common/currency.enum.js'
import { UserDto } from '../users/user.dto.js'
import { OrderItemDto, OrderProductDto } from './order-item.enum.js'
import { OrderStatus } from './order-status.enum.js'

export type OrderDto = {
  id: string
  userId: string
  status: OrderStatus
  currency: Currency
  items: OrderItemDto[]
  createdAt: string
  updatedAt: string
}

export type OrderWithProductDto = Omit<OrderDto, 'items'> & {
  products: OrderProductDto[]
}

export type CreateOrderDto = {
  userId: string
  currency: Currency
  items: OrderItemDto[]
}

export type OrderStatusUpdate = {
  orderId: string
  status: OrderStatus
}

export type OrderViewDto = Omit<OrderWithProductDto, 'userId'> & {
  user: UserDto
}

File: ecommerce-go-fiber/shared/contracts/orders/order_dto.go
package orders

import (
	"github.com/ivanmonichev/ecommerce-go-fiber/shared/contracts/common"
	"github.com/ivanmonichev/ecommerce-go-fiber/shared/contracts/users"
)

type OrderDTO struct {
	ID        string         `json:"id"`
	UserID    string         `json:"userId"`
	Status    OrderStatus    `json:"status"`
	Items     []OrderItemDTO `json:"items"`
	CreatedAt string         `json:"createdAt"`
	UpdatedAt string         `json:"updatedAt"`
}

type OrderWithProductDTO struct {
	ID        string            `json:"id"`
	UserID    string            `json:"userId"`
	Status    OrderStatus       `json:"status"`
	Currency  common.Currency   `json:"currency"`
	Products  []OrderProductDTO `json:"products"`
	CreatedAt string            `json:"createdAt"`
	UpdatedAt string            `json:"updatedAt"`
}

type CreateOrderDTO struct {
	UserID   string          `json:"userId"`
	Currency common.Currency `json:"currency"`
	Items    []OrderItemDTO  `json:"items"`
}

type OrderStatusUpdate struct {
	OrderID string      `json:"orderId"`
	Status  OrderStatus `json:"status"`
}

type OrderViewDTO struct {
	ID        string            `json:"id"`
	Status    OrderStatus       `json:"status"`
	Currency  common.Currency   `json:"currency"`
	Products  []OrderProductDTO `json:"products"`
	User      users.UserDTO     `json:"product"`
	CreatedAt string            `json:"createdAt"`
	UpdatedAt string            `json:"updatedAt"`
}
